### PR TITLE
Move dependency model from cli repo to core-setup

### PR DESCRIPTION
**Pull request by livarcocc**
cc @eerhardt @piotrpMSFT @brthor

### CODE REVIEW

**Comment by livarcocc** [comment]
@dotnet-bot Test RHEL7.2 x64 Release Build please.

File: build_projects/dotnet-host-build/CompileTargets.cs [hunk 1]
```diff
@@ -253,6 +254,34 @@ public static BuildTargetResult CompileCoreHost(BuildTargetContext c)
...
+                var configuration = c.BuildContext.Get<string>("Configuration");
+
+                // build projects for nuget packages
+                var packagingOutputDir = Path.Combine(Dirs.Intermediate, "forPackaging");
+                Mkdirp(packagingOutputDir);
+                foreach (var project in PackageTargets.ProjectsToPack)
+                {
+                    // Just build them, we'll pack later
+                    var packBuildResult = DotNetCli.Stage0.Build(
+                        "--build-base-path",
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/CompileTargets.cs]: (nit): it makes sense to me to put the 2 args that go together on the same line.  I think it is more readable.

```
"--build-base-path", packagingOutputDir,
"--configuration", configuration,
```

File: build_projects/dotnet-host-build/PackageTargets.cs [hunk 2]
```diff
@@ -13,6 +13,12 @@ namespace Microsoft.DotNet.Host.Build
 {
     public static class PackageTargets
     {
+        public static readonly string[] ProjectsToPack  = new string[]
+        {
+            "Microsoft.DotNet.PlatformAbstractions",
```
Comment (eerhardt) [inline build_projects/dotnet-host-build/PackageTargets.cs]: Should this be named "Microsoft.Extensions.PlatformAbstractions" to fit better with "Microsoft.Extensions.DependencyModel"?

File: test/HostActivationTests/project.json [hunk 3]
```diff
@@ -12,7 +12,8 @@
     "System.Runtime.InteropServices.RuntimeInformation": "4.0.0",
     "TestUtils": {
       "target": "project"
-    }
+    },
+    "Newtonsoft.Json": "9.0.1"
```
Comment (eerhardt) [inline test/HostActivationTests/project.json]: Why is this needed?  We shouldn't need to touch these tests.

File: test/TestUtils/project.json [hunk 4]
```diff
@@ -27,7 +27,8 @@
     "xunit": "2.1.0",
     "xunit.netcore.extensions": "1.0.0-prerelease-00206",
     "dotnet-test-xunit": "1.0.0-rc2-192208-24",
-    "System.Runtime.InteropServices.RuntimeInformation": "4.0.0"
+    "System.Runtime.InteropServices.RuntimeInformation": "4.0.0",
+    "Microsoft.DotNet.InternalAbstractions": "1.0.0-rc2-002702"
```
Comment (eerhardt) [inline test/TestUtils/project.json]: This is a pretty old version.  Is there something newer that we should be using?

**Comment by eerhardt** [comment]
Looks good.  I just had a few minor comments/questions.  And then one naming comment/question.

File: test/TestUtils/project.json (hunk 4 above)
Comment (livarcocc) [inline test/TestUtils/project.json]: I just used the one that was being used in other places in the repo.

File: test/HostActivationTests/project.json (hunk 3 above)
Comment (livarcocc) [inline test/HostActivationTests/project.json]: I hit a runtime exception without this.

**Comment by livarcocc** [comment]
Can't really call it Extensions.PlatformAbstractions because there is a package called that already. I will stick with the old name.

**Comment by eerhardt** [comment]
I see that now:  http://www.nuget.org/packages/Microsoft.Extensions.PlatformAbstractions/

But isn't this going to be confusing?  Won't customers ask what the difference is between:

```
Microsoft.DotNet.InternalAbstractions
Microsoft.DotNet.PlatformAbstractions
Microsoft.Extensions.PlatformAbstractions
```

?

**Comment by livarcocc** [comment]
Yes, but we are not exactly shipping this as an API to be used by customers, right? If we were, we should be spending A LOT, and I mean A LOT more time designing the APIs surface.

**Comment by eerhardt** [comment]
> Yes, but we are not exactly shipping this as an API to be used by customers, right?

It is a public API in an officially shipped NuGet package.  I think we should treat it just like any other public API.

**Comment by livarcocc** [comment]
Well, then we should stop picking and choosing what we pick to treat as any other public API. The very content of the package should then be improved tremendously. It is very confusing to me how we pick on the naming of the package but the contents are far from great. Why treat the naming with one standard but the contents of the package with another?

I imagined we were using the API as we are because it is not really intended to be a package like System.Linq.

File: test/TestUtils/project.json (hunk 4 above)
Comment (brthor) [inline test/TestUtils/project.json]: The only project I see using this is the Build Framework. That dependency isn't critical to be up to date, but don't you think our tests should be using something more recent?

**Comment by brthor** [comment]
LGTM :shipit:
